commcare_connect/reports/management/commands/backfill_user_analytics_data.py

```diff
@@ -0,0 +1,125 @@
+from collections import defaultdict
+from itertools import chain
+
+from django.core.management import BaseCommand
+from django.db.models import Case, Count, DateTimeField, F, IntegerField, Max, Min, Q, Sum, Value, When
+from django.db.models.lookups import GreaterThanOrEqual
+
+from commcare_connect.connect_id_client.main import fetch_user_analytics
+from commcare_connect.opportunity.models import CompletedWorkStatus, OpportunityAccess
+from commcare_connect.reports.models import UserAnalyticsData
+from commcare_connect.users.models import User
+
+
+class Command(BaseCommand):
+    help = "Backfills User Analytics Data"
+
+    def handle(self, *args, **options):
+        users = User.objects.filter(username__isnull=False, email__isnull=True, is_active=True).values_list(
+            "id", "username"
+        )
+        personalid_analytics_data = fetch_user_analytics()
+
+        access_objects = (
+            OpportunityAccess.objects.filter(user_id__in=[user_id for user_id, _ in users])
+            .annotate(
+                max_total=Sum("opportunity__paymentunit__max_total"),
+                max_total_done=Count(
+                    "completedwork__id", filter=Q(completedwork__status=CompletedWorkStatus.approved.value)
+                ),
+            )
+            .values("user_id")
+            .annotate(
+                total=Count("id", distinct=True),
+                accepted_count=Count("id", filter=Q(accepted=True), distinct=True),
+                completed_count=Sum(
+                    Case(
+                        When(Q(GreaterThanOrEqual("max_total_done", "max_total"), accepted=True), then=Value(1)),
+                        default=Value(0),
+                        output_field=IntegerField(),
+                    )
+                ),
+            )
+            .annotate(
+                username=F("user__username"),
+                has_opp_invite=Max("invited_date"),
```

Comment:
I think a lot of these dates are not quite right. They are picking the `MAX` (latest) date, not the earliest time it occurred. I would expect most to want to be `MIN` unless there is a specific reason around null values, but naively I would assume min excludes null anyway.